nasa/cumulus
# [CUMULUS-2475][CUMULUS-2476] GET and HEAD endpoints

**Summary:** Add GET and HEAD endpoints to distribution API.

Addresses 
- [CUMULUS-2475: Implement GET endpoint](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2475)
- [CUMULUS-2476: Implement HEAD endpoint](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2476)

## Changes

* CUMULUS-2475
Adds GET endpoint
To test: 
go to `<dist-api-url>/bucket/key` in browser
Public files should not require auth
Private files should require auth (will redirect you to Cognito)

* CUMULUS-2476
Adds HEAD Endpoint
To test:
1. Basic Request: 
```sh
curl --location --head '<dist-api-url>/path/to/science-file.hdr'
```
2. With Range Headers
```sh
curl --location --head '<dist-api-url>/path/to/science-file.hdr' \
--header 'Range: bytes=0-1023'
```
3. Basic Request (Authenticated)
```sh
curl --location --head '<dist-api-url>/path/to/science-file.hdr' \
--header 'Cookie: accessToken=<access-token-from-cognito>'
```
4. Range Requests (Authenticated)
```sh
curl --location --head '<dist-api-url>/path/to/science-file.hdr' \
--header 'Range: bytes=0-1023' \
--header 'Cookie: accessToken=<access-token-from-cognito>'
```   

## PR Checklist
